File: src/FirstHttpFunction.cs
using System.Net;
using System.Text.Json;
using Microsoft.Azure.Functions.Worker;
using Microsoft.Azure.Functions.Worker.Http;
using Microsoft.Extensions.Logging;

namespace FunctionApp;

public class FirstHttpFunction
{
    private readonly ILogger<FirstHttpFunction> _logger;
    private readonly IHttpClientFactory _httpClientFactory;

    public FirstHttpFunction(ILogger<FirstHttpFunction> logger, IHttpClientFactory httpClientFactory)
    {
        _logger = logger;
        _httpClientFactory = httpClientFactory;
    }

    [Function("first_http_function")]
    public async Task<HttpResponseData> Run(
        [HttpTrigger(AuthorizationLevel.Anonymous, "get", "post", Route = "first_http_function")] HttpRequestData req)
    {
        _logger.LogInformation("C# HTTP trigger function (first) processed a request.");

        // Call the second function
        var baseUrl = $"{req.Url.Scheme}://{req.Url.Authority}/api";
        var secondFunctionUrl = $"{baseUrl}/second_http_function";

        var httpClient = _httpClientFactory.CreateClient();
        var response = await httpClient.GetAsync(secondFunctionUrl);
        var secondFunctionResult = await response.Content.ReadAsStringAsync();

        var result = new
        {
            message = "Hello from the first function!",
            second_function_response = secondFunctionResult
        };

        var httpResponse = req.CreateResponse(HttpStatusCode.OK);
        httpResponse.Headers.Add("Content-Type", "application/json");
        await httpResponse.WriteStringAsync(JsonSerializer.Serialize(result));

        return httpResponse;
    }
}

File: src-ts/src/functions/firstHttpFunction.ts
import { HttpRequest, HttpResponseInit, InvocationContext } from '@azure/functions';

export async function firstHttpFunction(request: HttpRequest, context: InvocationContext): Promise<HttpResponseInit> {
    context.log('TypeScript HTTP trigger function (first) processed a request.');
    
    try {
        // Call the second function
        const baseUrl = request.url.split('/api/')[0];
        const secondFunctionUrl = `${baseUrl}/api/second_http_function`;
        
        const response = await fetch(secondFunctionUrl);
        const secondFunctionResult = await response.text();
        
        const result = {
            message: 'Hello from the first function!',
            second_function_response: secondFunctionResult
        };
        
        return {
            status: 200,
            headers: {
                'Content-Type': 'application/json'
            },
            body: JSON.stringify(result)
        };
    } catch (error) {
        context.error('Error calling second function:', error);
        return {
            status: 500,
            body: JSON.stringify({ error: 'Failed to call second function' })
        };
    }
}
